src/libopensc/simpletlv.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * simpletlv.h: Simple TLV header file
+ *
+ * Copyright (C) 2016  Red Hat, Inc., Robert Relyea <rrelyea@redhat.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef _OPENSC_SIMPLETLV_H
+#define _OPENSC_SIMPLETLV_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "libopensc/opensc.h"
+#include "libopensc/pkcs15.h"
+
+/*
+ * Create a tag/length file in Simple TLV based on the  val_len  content length
+ * @param  tag       Tag to store into the TL file
+ * @param  val_len   Value length to store into the TL file
+ * @param  tlp       TL byte array to write into (expects at most 4 bytes written)
+ * @param  tl_len_p  The amount of bytes written to the TL file
+ * @return           SC_SUCCESS for correct input
+ */
+int simpletlv_create_tag_length(u8 tag, size_t val_len, u8 *tlp, size_t *tl_len_p);
+
+/* get the Simple TLV length return that len and return the length of the TL
+ * header. Don't overrun the buffer.
+ * @param  tl_ptr      Pointer to the TL file
+ * @param  tl_len      The length of TL file
+ * @param  tl_head_len The length of parsed header
+ * @return             The length from TL file
+ */
+size_t simpletlv_get_len(u8 *tl_ptr, size_t tl_len, size_t *tl_head_len);
```

Comment:
Your simpletlv implementation has some limitations, which is OK. But could you please match the API to those of `sc_asn1_put_tag` and `sc_asn1_read_tag`?